File: 219. Contains Duplicate II/219.js
// 219. Contains Duplicate II solution in JavaScript
// https://leetcode.com/problems/contains-duplicate-ii/description/

/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
 var containsNearbyDuplicate = function(nums, k) {
    let set = new Set();

    for (index in nums) {
        if (set.has(nums[index])) return true;

        set.add(nums[index]);

        if (set.size > k) {
            set.delete(nums[index - k])
        }
    }

    return false;
};

File: 219. Contains Duplicate II/219.go
// 219. Contains Duplicate II solution in GoLang
// https://leetcode.com/problems/contains-duplicate-ii/description/

func containsNearbyDuplicate(nums []int, k int) bool {
    num_map := make(map[int]bool)

    for index, num := range nums {
        if num_map[num] == true {
            return true
        }

        num_map[num] = true

        if len(num_map) > k {
            delete(num_map, nums[index - k])
        }
    }

    return false
}